org.scala-ide.sdt.core/src/org/scalaide/core/completion/ProposalRelevanceCalculator.scala

```diff
@@ -0,0 +1,85 @@
+package org.scalaide.core.completion
+
+import scala.tools.nsc.interactive.Global
+import org.scalaide.logging.HasLogger
+import scala.util.matching.Regex
+
+object ProposalRelevanceCalculator {
+  private final val MaxInternalRelevance = 1000
+  private final val MaxExternalRelevance = 100
+  private final val IntToExtRatio = MaxInternalRelevance.toDouble / MaxExternalRelevance
+
+  private def internalToExternalRelevance(relevance: Int): Int = {
+    if (relevance <= 0) 0
+    else if (relevance >= MaxInternalRelevance) MaxExternalRelevance
+    else (relevance * IntToExtRatio).round.toInt
+  }
+}
+
+class ProposalRelevanceCalculator(cfg: ProposalRelevanceCfg = DefaultProposalRelevanceCfg) extends HasLogger {
+  import ProposalRelevanceCalculator._
+
+  def forScala[CompilerT <: Global](pc: CompilerT)(prefix: String, name: String, sym: pc.Symbol, viaView: pc.Symbol, inherited: Option[Boolean]): Int = {
+    // rudimentary relevance, place own members before inherited ones, and before view-provided ones
+    var relevance = MaxInternalRelevance
```

Comment:
You are right, the code is not from me, but giving all this magic numbers proper names would make sense indeed. I can take care of this in the days to come...